### PR TITLE
Enable retry logic integration tests

### PR DESCRIPTION
**Pull request by lemillermicrosoft**
### Motivation and Context
#420 skipped retry tests for OpenAI. It was noted that Azure SDK was an issue, however, I added a test for Azure OpenAI and it passes as expected as well. I know there are changes coming in the area. I'd like to re-enable the tests since they are passing to make sure functionality remains intact as design changes.

### Description
Enable the previous test and clone for Azure OpenAI.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:
- [x] [Passing integration tests](https://github.com/lemillermicrosoft/semantic-kernel/actions/runs/5021590753)
image
image

### CODE REVIEW

**Review by RogerBarreto — DISMISSED** [review]

File: dotnet/src/IntegrationTests/Connectors/OpenAI/OpenAICompletionTests.cs [hunk 1]
```diff
@@ -105,7 +105,7 @@ public async Task AzureOpenAITestAsync(bool useChatModel, string prompt, string
         Assert.Contains(expectedAnswerContains, actual.Result, StringComparison.OrdinalIgnoreCase);
     }
 
-    [Theory(Skip = "Retry logic needs to be refactored to work with Azure SDK")]
+    [Theory]
```
Comment (dluc) [inline dotnet/src/IntegrationTests/Connectors/OpenAI/OpenAICompletionTests.cs:108]: has the retry logic been fixed/refactored?

Comment (lemillermicrosoft) [reply dotnet/src/IntegrationTests/Connectors/OpenAI/OpenAICompletionTests.cs:108]: It's in progress no, but things are already working. I'm not sure what changed when, but I know that around the time of #420 integration tests were red across the board so understandable that we lost that trail.

Comment (lemillermicrosoft) [inline dotnet/src/IntegrationTests/Connectors/OpenAI/OpenAICompletionTests.cs:108]:
```suggestion
    // If the test fails, please note that SK retry logic may not be fully integrated into the underlying code using Azure SDK
    [Theory]
```

File: dotnet/src/IntegrationTests/Connectors/OpenAI/OpenAICompletionTests.cs [hunk 2]
```diff
@@ -132,6 +132,34 @@ public async Task OpenAIHttpRetryPolicyTestAsync(string prompt, string expectedO
         Assert.Contains(expectedOutput, this._testOutputHelper.GetLogs(), StringComparison.OrdinalIgnoreCase);
     }
 
+    [Theory]
```
Comment (lemillermicrosoft) [inline dotnet/src/IntegrationTests/Connectors/OpenAI/OpenAICompletionTests.cs:135]:
```suggestion
    // If the test fails, please note that SK retry logic may not be fully integrated into the underlying code using Azure SDK
    [Theory]
```

**Review by RogerBarreto — APPROVED** [review]

**Pull request merged**